crates/pathfinder/src/state/sync.rs

```diff
@@ -440,81 +383,6 @@ async fn update_sync_status_latest(
     }
 }
 
-async fn l1_update(connection: &mut Connection, updates: &[StateUpdateLog]) -> anyhow::Result<()> {
-    tokio::task::block_in_place(move || {
-        let transaction = connection
-            .transaction_with_behavior(TransactionBehavior::Immediate)
-            .context("Create database transaction")?;
-
-        for update in updates {
-            L1StateTable::upsert(&transaction, update).context("Insert update")?;
-        }
-
-        // Track combined L1 and L2 state.
-        let l1_l2_head = RefsTable::get_l1_l2_head(&transaction).context("Query L1-L2 head")?;
-        let expected_next = l1_l2_head
-            .map(|head| head + 1)
-            .unwrap_or(StarknetBlockNumber::GENESIS);
-
-        match updates.first() {
-            Some(update) if update.block_number == expected_next => {
-                let mut next_head = None;
-                for update in updates {
-                    let l2_root = StarknetBlocksTable::get_state_commitment(
-                        &transaction,
-                        update.block_number.into(),
-                    )
-                    .context("Query L2 root")?
-                    .map(|(a, b)| StateCommitment::calculate(a, b));
-
-                    match l2_root {
-                        Some(l2_root) if l2_root == update.global_root => {
-                            next_head = Some(update.block_number);
-                        }
-                        _ => break,
-                    }
-                }
-
-                if let Some(next_head) = next_head {
-                    RefsTable::set_l1_l2_head(&transaction, Some(next_head))
-                        .context("Update L1-L2 head")?;
-                }
-            }
-            _ => {}
-        }
-
-        transaction.commit().context("Commit database transaction")
-    })
-}
-
-async fn l1_reorg(
-    connection: &mut Connection,
-    reorg_tail: StarknetBlockNumber,
-) -> anyhow::Result<()> {
-    tokio::task::block_in_place(move || {
-        let transaction = connection
-            .transaction_with_behavior(TransactionBehavior::Immediate)
-            .context("Create database transaction")?;
-
-        L1StateTable::reorg(&transaction, reorg_tail).context("Delete L1 state from database")?;
-
-        // Track combined L1 and L2 state.
-        let l1_l2_head = RefsTable::get_l1_l2_head(&transaction).context("Query L1-L2 head")?;
```

Comment:
iiuc the L1 state table manages an L1 reorg internally by doing:

```
tx.execute(
            r"DELETE FROM l1_state WHERE 
            starknet_block_number >= :starknet_block_number OR 
            ethereum_block_number >= :ethereum_block_number",
            named_params! {
                ":starknet_block_number": update.block_number,
                ":ethereum_block_number": update.eth_block_number,
            },
        )?;
```

just before inserting the latest fetched L1 state but I don't see a `RefsTable::set_l1_l2_head` which means the l1/l2 head would contain an invalid value

Reply:
having looked in the code the only side effect now is that following an L1 reorg the l1/l2 head is not updated so some L2 blocks would be reporting L1 instead of L2 accepted status

Reply:
Done, L1 head is updated per-block and effectively tracks L1 head for most recent downloaded block.